packages/main/src/CalendarLegend.ts

```diff
@@ -0,0 +1,131 @@
+import UI5Element from "@ui5/webcomponents-base/dist/UI5Element.js";
+import customElement from "@ui5/webcomponents-base/dist/decorators/customElement.js";
+import litRender from "@ui5/webcomponents-base/dist/renderer/LitRenderer.js";
+import property from "@ui5/webcomponents-base/dist/decorators/property.js";
+import ItemNavigation, { ITabbable } from "@ui5/webcomponents-base/dist/delegate/ItemNavigation.js";
+import ItemNavigationBehavior from "@ui5/webcomponents-base/dist/types/ItemNavigationBehavior.js";
+import CalendarLegendItemType from "./types/CalendarLegendItemType.js";
+import CalendarLegendTemplate from "./generated/templates/CalendarLegendTemplate.lit.js";
+
+// Styles
+import CalendarLegendCss from "./generated/themes/CalendarLegend.css.js";
+import CalendarLegendItem from "./CalendarLegendItem.js";
+
+/**
+ * @class CalendarLegend
+ *
+ * <h3 class="comment-api-title">Overview</h3>
+ *
+ * The <code>ui5-calendar-legend</code> component is used to display a legend of special calendar dates
+ * with their corresponding textual descriptions and types.
+ *
+ * <h3>ES6 Module Import</h3>
+ *
+ * <code>import "@ui5/webcomponents/dist/CalendarLegend.js";</code>
+ *
+ * @constructor
+ * @author SAP SE
+ * @alias sap.ui.webc.main.CalendarLegend
+ * @extends sap.ui.webc.base.UI5Element
+ * @tagname ui5-calendar-legend
+ * @public
+ */
+@customElement({
+	tag: "ui5-calendar-legend",
+	renderer: litRender,
+	styles: CalendarLegendCss,
+	template: CalendarLegendTemplate,
+	dependencies: [CalendarLegendItem],
+})
+class CalendarLegend extends UI5Element {
+	/**
+	 * Hides the Today item in the legend.
+	 *
+	 * @type {boolean}
+	 * @name sap.ui.webc.main.CalendarLegend.prototype.hideToday
+	 * @public
+	 */
+	@property({ type: Boolean })
+	hideToday!: boolean;
+
+	/**
+	 * Hides the Selected day item in the legend.
+	 *
+	 * @type {boolean}
+	 * @name sap.ui.webc.main.CalendarLegend.prototype.hideSelectedDay
+	 * @public
+	 */
+	@property({ type: Boolean })
+	hideSelectedDay!: boolean;
+
+	/**
+	 * Hides the Non-Working day item in the legend.
+	 *
+	 * @type {boolean}
+	 * @name sap.ui.webc.main.CalendarLegend.prototype.hideNonWorkingDay
+	 * @public
+	 */
+	@property({ type: Boolean })
+	hideNonWorkingDay!: boolean;
+
+	/**
+	 * Hides the Working day item in the legend.
+	 *
+	 * @type {boolean}
+	 * @name sap.ui.webc.main.CalendarLegend.prototype.hideWorkingDay
+	 * @public
+	 */
+	@property({ type: Boolean })
+	hideWorkingDay!: boolean;
+
+	_itemNavigation: ItemNavigation;
+
+	constructor() {
+		super();
+
+		this._itemNavigation = new ItemNavigation(this, {
+			getItemsCallback: () => this._getFocusableElements(),
+			behavior: ItemNavigationBehavior.Cyclic,
+		});
+	}
+
+	_onItemClick(e: MouseEvent) {
+		const target = e.target as CalendarLegendItem;
+
+		this._itemNavigation.setCurrentItem(target);
+	}
+
+	_itemsInLegend() {
+		const items = this.shadowRoot!.querySelectorAll<CalendarLegendItem>("ui5-calendar-legend-item");
```

Comment:
query on the tag of our components is not safe because of the component scoping, this could become demo-ui5-calendar-legend-item and then this method will return empty array. That's why we always set an attribute with the original name and always use the attribute selector